app/controllers/categories_controller.rb

```diff
@@ -9,4 +9,19 @@ def index
     categories = categories.where(top_level: top_level) unless top_level.nil?
     render json: CategoryPresenter.present(categories)
   end
+
+  # rubocop:disable Metrics/AbcSize
+  def counts
+    if !admin_signed_in?
+      render status: :unauthorized
+    else
+      render status: :ok, json:
+          Category.order(:name).map { |c|
```

Comment:
I think you want to use `do/end` instead of curly braces here. There's actually a difference between the two when it comes to syntax and binding strength: https://gemfile.wordpress.com/2014/04/04/do-end-vs-not-just-a-matter-of-style/

Reply:
That actually didn't work, will require more reading, so i'll leave as is for now.